### ISSUE TITLE
Can't compile "Seq(Vector(2), List(3))" in Scala 2.13.0-M4

### ISSUE THREAD

**Issue by xuwei-k** [scala/bug#10907]
```
Welcome to Scala 2.13.0-M4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_172).
Type in expressions for evaluation. Or try :help.

scala> Seq(Vector(2), List(3))
                 ^
       error: type mismatch;
        found   : scala.collection.immutable.Vector[Int]
        required: scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq with Serializable,Serializable with Equals] with Serializable,scala.collection.immutable.AbstractSeq[Any] with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq with Serializable,Serializable with Equals] with Serializable] with Serializable,scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq with Serializable,Serializable with Equals] with Serializable,scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq with Serializable,scala.collection.immutable.AbstractSeq[Int] with Serializable] with Serializable] with Serializable] with Serializable
                          ^
       error: type mismatch;
        found   : List[Int]
        required: scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq with Serializable,Serializable with Equals] with Serializable,scala.collection.immutable.AbstractSeq[Any] with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq with Serializable,Serializable with Equals] with Serializable] with Serializable,scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq with Serializable,Serializable with Equals] with Serializable,scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq with Serializable,scala.collection.immutable.AbstractSeq[Int] with Serializable] with Serializable] with Serializable] with Serializable
```

same result latest 2.13.x

```
resolvers += "scala-pr" at "https://scala-ci.typesafe.com/artifactory/scala-integration/"

scalaVersion := "2.13.0-pre-9e35986" // latest version 2018-05-26
```

```
Welcome to Scala 2.13.0-pre-9e35986 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_172).
Type in expressions for evaluation. Or try :help.

scala> Seq(Vector(2), List(3))
                 ^
       error: type mismatch;
        found   : scala.collection.immutable.Vector[Int]
        required: scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq,Serializable with Equals],scala.collection.immutable.AbstractSeq[Any] with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq,Serializable with Equals]],scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq,Serializable with Equals],scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq,scala.collection.immutable.AbstractSeq[Int]]]]
                          ^
       error: type mismatch;
        found   : List[Int]
        required: scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq,Serializable with Equals],scala.collection.immutable.AbstractSeq[Any] with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq,Serializable with Equals]],scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq with scala.collection.immutable.StrictOptimizedSeqOps[Any,scala.collection.immutable.AbstractSeq,Serializable with Equals],scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,scala.collection.immutable.AbstractSeq,scala.collection.immutable.AbstractSeq[Int]]]]
```

**Comment by xuwei-k** [comment]
not bug ?🤔  https://github.com/scala/scala/commit/29406158c9051475227d240876b36a8605b98c0e#diff-bfcc30d9cea7eef4f037fec586af6ca5R9

**Comment by xuwei-k** [comment]
similar issue https://github.com/scala/bug/issues/10908

**Comment by joroKr21** [comment]
scala/scala#6352 - I believe it's a bug. In Dotty intersection and union types make this easier.

**Comment by lrytz** [comment]
It is a bug, in the compiler (not the collections library). With the new collections, some expressions that used to work now trip over the compiler. However, some that failed on 2.12 now work on 2.13.

On 2.12 (https://github.com/scala/scala/blob/2.12.x/test/files/neg/lub-from-hell-2.scala)
```
scala> import collection.mutable._
import collection.mutable._

scala> List(Set(), ListBuffer())
<console>:14: error: type arguments [Nothing,Iterable[Nothing] with Nothing => Boolean with scala.collection.generic.Subtractable[Nothing,Iterable[Nothing] with Nothing => Boolean with scala.collection.generic.Subtractable[Nothing,Iterable[Nothing] with Nothing => Boolean]{def seq: Iterable[Nothing] with Nothing => Boolean}]{def seq: Iterable[Nothing] with Nothing => Boolean{def seq: Iterable[Nothing] with Nothing => Boolean}}] do not conform to trait Subtractable's type parameter bounds [A,+Repr <: scala.collection.generic.Subtractable[A,Repr]]
       val res0 =
           ^
<console>:15: error: type arguments [Nothing,Iterable[Nothing] with Nothing => Boolean with scala.collection.generic.Subtractable[Nothing,Iterable[Nothing] with Nothing => Boolean with scala.collection.generic.Subtractable[Nothing,Iterable[Nothing] with Nothing => Boolean]{def seq: Iterable[Nothing] with Nothing => Boolean}]{def seq: Iterable[Nothing] with Nothing => Boolean{def seq: Iterable[Nothing] with Nothing => Boolean}}] do not conform to trait Subtractable's type parameter bounds [A,+Repr <: scala.collection.generic.Subtractable[A,Repr]]
       List(Set(), ListBuffer())
       ^
```

This one "works" on 2.13 (gives one of those huge lubs).

**Comment by xuwei-k** [comment]
Before I knew it, fixed in latest 2.13 🤔 

```
Welcome to Scala 2.13.0-pre-8f1f22d (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_181).
Type in expressions for evaluation. Or try :help.

scala> Seq(Vector(2), List(3))
res0: Seq[scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,[_]scala.collection.immutable.AbstractSeq[_] with scala.collection.immutable.StrictOptimizedSeqOps[_,[_]scala.collection.immutable.AbstractSeq[_] with scala.collection.immutable.StrictOptimizedSeqOps[_,[_]scala.collection.immutable.AbstractSeq[_],scala.collection.immutable.AbstractSeq[_]],scala.collection.immutable.AbstractSeq[_] with scala.collection.immutable.StrictOptimizedSeqOps[_,[_]scala.collection.immutable.AbstractSeq[_],scala.collection.immutable.AbstractSeq[_]]],scala.collection.immutable.AbstractSeq[Int] with scala.collection.immutable.StrictOptimizedSeqOps[Int,[_]scala.collection.immutable.AbstractSeq[_] with scala.collection.immutable.StrictOptimizedSeqOps[_,[_]sc...
```

**Comment by xuwei-k** [comment]
https://github.com/scala/scala/pull/7043
